gui/webdisplay/src/RWebWindowsManager.cxx

```diff
@@ -159,26 +159,42 @@ void RWebWindowsManager::SetUseSessionKey(bool on)
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////
-/// Static method to generate kryptographic key
+/// Static method to generate cryptographic key
+/// Parameter keylen defines length of cryptographic key in bytes
+/// Output string will be hex formatted and includes "-" separator after every 4 bytes
+/// Example for 16 bytes: "fca45856-41bee066-ff74cc96-9154d405"
 
 std::string RWebWindowsManager::GenerateKey(int keylen)
 {
-   // try to randomize??
-   gRandom->SetSeed();
+   std::vector<unsigned char> buf(keylen, 0);
+   if (gSystem->GetCryptoRandom(buf.data(), keylen) != keylen) {
+      R__LOG_ERROR(WebGUILog()) << "Error in gSystem->GetCryptoRandom";
+      gSystem->Abort(11); // this line should abort ROOT application
+      // this code just to show old implementation
+      gRandom->SetSeed();
+      for (int n = 0; n < keylen; n++)
+         buf[n] = (unsigned char)gRandom->Integer(0xFF);
```

Comment:
I don't think we should keep dead code in the code base. The old implementation remains anyway available in git.